### PR TITLE
Get protos by trait info

### PR DESCRIPTION
**Pull request by alebaffa**
This PR increases the granularity of `get_protos` in case the category in input is `Categories::Trait(ShardsTraitInfo)`.
It makes it possible to search Protos by `ShardsTraitInfo.name` or `ShardsTraitInfo.id` using these patterns:
- if the `Category::Trait` in input contains `ShardsTraitInfo.id = [0u8; 16]`, then `get_protos` will search by `ShardsTraitInfo.name`
- if the `Category::Trait` in input contains `ShardsTraitInfo.name = ""`, then it will search by `ShardsTraitInfo.id`

### CODE REVIEW

**Comment by alebaffa** [comment]
Javascript tests failing. Need to check them.

**Comment by alebaffa** [comment]
It was a random error.

File: pallets/fragments/src/lib.rs [hunk 1]
```diff
@@ -442,8 +442,12 @@ pub mod pallet {
 			// create vault account
 			// we need an existential amount deposit to be able to create the vault account
 			let vault = Self::get_vault_id(hash);
-			let min_balance = <T as pallet_assets::Config>::Currency::minimum_balance();
-			<T as pallet_assets::Config>::Currency::reserve(&vault, min_balance)?;
+			let min_balance =
```
Comment (sinkingsugar) [inline pallets/fragments/src/lib.rs:445]: this is unrelated no?
Also @aang114 fixed it and this would also not work iirc?

File: pallets/fragments/src/lib.rs [hunk 2]
```diff
@@ -955,10 +959,10 @@ pub mod pallet {
 			let frag_account = Self::get_fragment_account_id(class, edition, copy);
 			let min_balance =
 				<pallet_balances::Pallet<T> as Currency<T::AccountId>>::minimum_balance();
-			<pallet_balances::Pallet<T> as ReservableCurrency<T::AccountId>>::reserve(
+			let _ = <pallet_balances::Pallet<T> as Currency<T::AccountId>>::deposit_creating(
```
Comment (sinkingsugar) [inline pallets/fragments/src/lib.rs:962]: @aang114 double check with you

File: pallets/protos/src/lib.rs [hunk 3]
```diff
@@ -251,6 +251,7 @@ pub mod pallet {
 		StakeLocked,
 		/// Cannot find FRAG link to use as stake funds
 		NoFragLink,
+		OwnerNotFound,
```
Comment (sinkingsugar) [inline pallets/protos/src/lib.rs]: hmm we generally use those errors for extrinsics. Maybe we need a new struct?

**Review by sinkingsugar — CHANGES_REQUESTED** [review]
LGTM

Just I noticed some unrelated code etc

File: pallets/fragments/src/lib.rs (hunk 1 above)
Comment (alebaffa) [reply pallets/fragments/src/lib.rs:445]: Yes, it's unrelated. I think it came up merging locally my branch with devel.

File: pallets/fragments/src/lib.rs (hunk 2 above)
Comment (alebaffa) [reply pallets/fragments/src/lib.rs:962]: Also this one.

File: pallets/protos/src/lib.rs (hunk 3 above)
Comment (alebaffa) [reply pallets/protos/src/lib.rs]: Oh, this can be removed actually. I am not using it.

File: pallets/fragments/src/lib.rs (hunk 1 above)
Comment (sinkingsugar) [reply pallets/fragments/src/lib.rs:445]: Weird if was merged should not show though 🤔

Comment (alebaffa) [reply pallets/fragments/src/lib.rs:445]: Weird, agreed. Trying to figure out how to remove these two files from this PR (which are already on devel).

**Comment by alebaffa** [comment]
Let me quickly do another PR and close this one. I don't want to mess things up for just three files...
